lib/session.js

```diff
@@ -693,12 +703,6 @@ Session.prototype._get_link = function (frame) {
 
 Session.prototype.on_detach = function (frame) {
     this._get_link(frame).on_detach(frame);
-    //remove link
-    var handle = frame.performative.handle;
-    var link = this.remote.handles[handle];
-    delete this.remote.handles[handle];
-    delete this.local.handles[link.local.handle];
-    delete this.links[link.name];
```

Comment:
Is there a reason to not remove the link on detach?

Reply:
It should not be removed until the event has been handled by the application (if it wants to) and the local close is issued.